corehq/apps/reports/standard/monitoring.py

```diff
@@ -1799,16 +1803,32 @@ def rows(self):
         return rows
 
 
-def _get_raw_user_link(user, url, filter_class):
+def _get_raw_user_link(user, url, filter_class, additional_params=None):
     """
-    filter_class is expected to be either ExpandedMobileWorkerFilter or a
-    subclass of it, such as the CaseListFilter
+    Generates an HTML anchor tag (<a>) for a user link with additional query parameters.
+
+    Parameters:
+    - user (SimplifiedUserInfo object): The user for whom the link is being generated.
+    - url (str): The base URL to which filter parameters and any additional parameters will be appended.
+    - filter_class (class): A filter class expected to be either ExpandedMobileWorkerFilter or a subclass of it,
+    such as CaseListFilter.
+    - additional_params (dict, optional): An optional dictionary of additional query parameters to be appended to
+    the URL. Each key-value pair in the dictionary represents a parameter name and its value.
+
+    Returns:
+    - str: A string containing an HTML anchor tag (<a>) with the constructed URL and the user's display name.
     """
     user_link_template = '<a href="{link}?{params}">{username}</a>'
+    additional_param_string = ''
+    if additional_params:
+        if isinstance(additional_params, dict):
+            additional_param_string = '&' + urlencode(additional_params)
+        else:
+            raise ValueError("additional_params must be a dictionary")
     user_link = format_html(
         user_link_template,
         link=url,
-        params=urlencode(filter_class.for_user(user.user_id)),
+        params=urlencode(filter_class.for_user(user.user_id)) + additional_param_string,
```

Comment:
Following on the comment above, another option would be to do a single `urlencode`.
```suggestion
        params=urlencode(filter_class.for_user(user.user_id) | (additional_params or {})),
```
This suggestion would replace the `additional_param_string` and the `if additional_params: ...` block above.